File: src/main/java/com/example/indie91/Models/ProductMedia.java
package com.example.indie91.Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "product_product_media")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductMedia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // This is fine as `int8` maps to `Long`

    @Column(name = "product_id", columnDefinition = "uuid")
    private UUID productId;

    @Column(name = "product_media_id", columnDefinition = "uuid")
    private UUID mediaId;
}

File: models/product_media.go
package models

type ProductMedia struct {
	ID    string `gorm:"column:id;primaryKey;type:uuid;default:gen_random_uuid()" json:"id"`
	URL   string `gorm:"column:url" json:"url"`

	ProductProductMedia []ProductProductMedia `gorm:"foreignKey:ProductMediaID;references:ID" json:"-"`
}

func (ProductMedia) TableName() string {
	return "product_media"
}
